acceptance/analyzer_test.go

```diff
@@ -76,317 +77,500 @@ func TestAnalyzer(t *testing.T) {
+					analyzeImage,
+					h.WithFlags("--env", "CNB_PLATFORM_API="+apiString),
+					h.WithArgs(
+						ctrPath(analyzerPath),
+						"some-image",
+					),
+				)
 
-			assertAnalyzedMetadata(t, filepath.Join(copyDir, "some-analyzed.toml"))
+				expected := "Not restoring cached layer metadata, no cache flag specified."
+				h.AssertStringContains(t, output, expected)
+			})
 		})
-	})
-
-	when("daemon case", func() {
-		it("writes analyzed.toml", func() {
-			h.DockerRunAndCopy(t,
-				containerName,
-				copyDir,
-				ctrPath("/layers/analyzed.toml"),
-				analyzeImage,
-				h.WithFlags(append(
-					dockerSocketMount,
-					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
-				)...),
-				h.WithArgs(
-					ctrPath(analyzerPath),
-					"-daemon",
-					"some-image",
-				),
-			)
 
-			assertAnalyzedMetadata(t, filepath.Join(copyDir, "analyzed.toml"))
+		when("the provided layers directory isn't writeable", func() {
+			it("recursively chowns the directory", func() {
+				h.SkipIf(t, runtime.GOOS == "windows", "Not relevant on Windows")
+
+				imgArg := "some-image"
+				// Platform >= 0.7 does not allow an argument, < 7 requires an argument
+				if api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0 {
+					imgArg = ""
+				}
+
+				output := h.DockerRun(t,
+					analyzeImage,
+					h.WithFlags("--env", "CNB_PLATFORM_API="+apiString),
+					h.WithBash(fmt.Sprintf("chown -R 9999:9999 /layers; chmod -R 775 /layers; %s %s; ls -al /layers", analyzerPath, imgArg)),
+				)
+
+				h.AssertMatch(t, output, "2222 3333 .+ \\.")
+				h.AssertMatch(t, output, "2222 3333 .+ group.toml")
+			})
 		})
 
-		when("app image exists", func() {
-			var appImage string
+		when("group path is provided", func() {
+			it("uses the provided group path", func() {
+				h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not accept a -group flag")
+				h.DockerSeedRunAndCopy(t,
+					containerName,
+					cacheFixtureDir, ctrPath("/cache"),
+					copyDir, ctrPath("/layers"),
+					analyzeImage,
+					h.WithFlags(
+						"--env", "CNB_PLATFORM_API="+apiString,
+					),
+					h.WithArgs(
+						ctrPath(analyzerPath),
+						"-cache-dir", ctrPath("/cache"),
+						"-group", ctrPath("/layers/other-group.toml"),
+						"some-image",
+					),
+				)
 
-			it.Before(func() {
-				appImage = "some-app-image-" + h.RandString(10)
-				metadata := minifyMetadata(t, filepath.Join("testdata", "analyzer", "app_image_metadata.json"), platform.LayersMetadata{})
+				h.AssertPathExists(t, filepath.Join(copyDir, "layers", "some-other-buildpack-id"))
+				h.AssertPathDoesNotExist(t, filepath.Join(copyDir, "layers", "some-buildpack-id"))
+			})
+		})
 
-				cmd := exec.Command(
-					"docker",
-					"build",
-					"-t", appImage,
-					"--build-arg", "fromImage="+containerBaseImage,
-					"--build-arg", "metadata="+metadata,
-					filepath.Join("testdata", "analyzer", "app-image"),
+		when("analyzed path is provided", func() {
```

Comment:
Can we replace the two `it` under this `when` to one by adding an `if` condition based on the `platform API`?

Reply:
6d7c623